tidy3d/components/simulation.py

```diff
@@ -1810,11 +1810,15 @@ def _filter_structures_plane(  # pylint:disable=too-many-locals
         for structure in structures:
 
             # dont bother with geometries that dont intersect plane
-            if not structure.geometry.intersects_plane(x=x, y=y, z=z):
+            if not structure.geometry.intersects(plane):
```

Comment:
I think this reads like a `Geometry.intersections(plane)` method yea.

So right now we have
```py
Geometry.intersections(x,y,z) -> List[Shapely]
Geometry.intersects_plane(x,y,z) -> bool
```
maybe we should in fact introduce a couple analogues that accept Box as arguments.
Not sure what to name them, but if we could start over, I'd say
```py
Geometry.intersections(x,y,z) -> List[Shapely]
Geometry.intersections_plane(plane: Box) -> List[Shapely]
Geometry.intersects(x,y,z) -> bool
Geometry.intersects_plane(plane: Box) -> bool
```
although this breaks backwards compatibility.
Maybe for the plane accepting ones, we could do `Geometry.interesects_plane_box` and `Geometry.intersections_box`. For example?